client/src/telethon/_impl/mtsender/sender.py

```diff
@@ -301,12 +273,13 @@ def _try_fill_write(self) -> None:
         result = self._mtp.finalize()
         if result:
             container_msg_id, mtp_buffer = result
+
+            self._transport.pack(mtp_buffer, self._writer.write)
             for request in self._requests:
                 if isinstance(request.state, Serialized):
-                    request.state.container_msg_id = container_msg_id
+                    request.state = Sent(request.state.msg_id, container_msg_id)
 
-            self._transport.pack(mtp_buffer, self._writer.write)
-            self._write_drain_pending = True
+            await self._writer.drain()
```

Comment:
Likewise, this blocks until a request it sent.

Reply:
I put it by mistake, I'll remove it